db/migrate/20141215104802_migrate_attachments_to_carrier_wave.rb

```diff
@@ -0,0 +1,129 @@
+      migrate_attachment attachment
+    end
+
+    Rails.logger.info 'Attachment migration complete.'
+  end
+
+  def down
+    count = Attachment.count
+
+    Rails.logger.info "Rolling back #{count} attachments from CarrierWave to \
+                       legacy file-based storage.".squish
+    Rails.logger.warn 'Depending on your configuration this can take a while.
+                       Especially if files are downloaded from S3.'.squish
+
+    Attachment.all.each_with_index do |attachment, i|
+      Rails.logger.info "Migrating attachment #{i + 1}/#{count} (#{attachment.file.path})"
+      rollback_attachment attachment
+    end
+
+    remove_column :attachments, :file
+  end
+
+  ##
+  # In the olden times (pre 4.1) files were all saved under the configured storage path,
+  # most likely `files`, with a timestamp attached to their name to avoid conflicts.
+  #
+  # e.g. `files/140917231758_openshift-setup.txt`
+  #
+  # The new approach is using CarrierWave for attachments. There the files will be stored
+  # some place else. Where is not material for this migration.
+  # It could be another directory or S3.
+  #
+  # The migration does the following:
+  #
+  # * rename the legacy files - it strips off the timestamps
+  # * assign the renamed file to the CarrierWave uploader - this moves the file to
+  #   the new location
+  #
+  # The rollback does the opposite using the remaining legacy column `disk_filename`
+  # of Attachment. Meaning it adds the timestamp again and puts the files back into
+  # the configured attachment storage path.
+  #
+  # This assumes that the attachment storage path is the same as before the migration
+  # to CarrierWave and also that the legacy columns `filename` and `disk_filename`
+  # are still present.
+  module LegacyAttachment
+    def migrate_attachment(attachment)
+      file = legacy_file_name attachment.disk_filename
+      new_file = strip_timestamp_from_filename(file)
+
+      if File.readable? file
+        FileUtils.move file, new_file
+        attachment.file = File.open(new_file)
+        attachment.save!
+        attachment.update_column :filename, ''
+
+        FileUtils.rm_f new_file
+
+        File.readable? attachment.file.path
+      else
+        if File.readable? attachment.file.path
+          true # file has been migrated already
+        else
+          Rails.logger.warn "Found corrupt attachment during migration: #{attachment.inspect}"
+          false
+        end
+      end
+    end
+
+    def rollback_attachment(attachment)
+      return unless attachment.file.path
+
+      old_file = rolled_back_file_name attachment
+
+      unless File.readable? old_file
+        file = attachment.diskfile.path
+
+        FileUtils.move file, old_file
+        attachment.update_column :file, nil
+        attachment.update_column :filename, Pathname(file).basename.to_s
+        attachment.update_column :disk_filename, Pathname(old_file).basename.to_s
```

Comment:
Then either use update_attributes or if you want to circumvent validation update_all(...).where(id: attachment.id)